extensions/vertx/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java

```diff
@@ -546,14 +546,15 @@ public Object captureContext() {
 
             @Override
             public void runWith(Runnable task, Object context) {
-                if (context != null) {
+                ContextInternal currentContext = (ContextInternal) Vertx.currentContext();
+                if (context != null && context != currentContext) {
```

Comment:
And now the philosophical question: should we distinguish root context and duplicated context? 

I do not believe so, but I think we should consider the question.

Reply:
Duplicated contexts do not share the context-local data, so I would say we are running with a _different_ context.